src/framework/vst/devtools/vstpluginlistmodelexample.h

```diff
@@ -23,54 +23,93 @@
+    Q_PROPERTY(QVariantList trackIdList READ trackIdList WRITE setTrackIdList NOTIFY trackIdListChanged)
+    Q_PROPERTY(QVariantList availableFxResources READ availableFxResources NOTIFY availableFxResourcesChanged)
+    Q_PROPERTY(QVariantList availableSynthResources READ availableSynthResources NOTIFY availableSynthResourcesChanged)
+
+    INJECT(vst, audio::IPlayback, playback)
     INJECT(vst, framework::IInteractive, interactive)
-    INJECT(vst, IVstPluginRepository, repository)
 public:
     explicit VstPluginListModelExample(QObject* parent = nullptr);
-    ~VstPluginListModelExample();
 
     Q_INVOKABLE void load();
-    Q_INVOKABLE void showPluginEditor();
+    Q_INVOKABLE void showSynthPluginEditor();
+    Q_INVOKABLE void showFxPluginEditor();
 
-    int rowCount(const QModelIndex& parent = QModelIndex()) const override;
-    QVariant data(const QModelIndex& index, int role) const override;
-    QHash<int, QByteArray> roleNames() const override;
+    const QVariantList& sequenceIdList() const;
+    void setSequenceIdList(const QVariantList& newSequenceIdList);
 
-    int selectedItemIndex() const;
+    const QVariantList& trackIdList() const;
+    void setTrackIdList(const QVariantList& newTrackIdList);
 
-public slots:
-    void setSelectedItemIndex(int selectedItemIndex);
+    const QVariantList& availableFxResources() const;
+    void setAvailableFxResources(const QVariantList& newAvailableFxResources);
 
-signals:
-    void selectedItemIndexChanged(int selectedItemIndex);
+    const QVariantList& availableSynthResources() const;
+    void setAvailableSynthResources(const QVariantList& newAvailableSynthResources);
 
-private:
-    enum Roles {
-        NameRole = Qt::UserRole + 1,
-        IdRole,
-        PathRole
-    };
+    int currentSequenceId() const;
+    void setCurrentSequenceId(int newCurrentSequenceId);
+
+    int currentTrackId() const;
+    void setCurrentTrackId(int newCurrentTrackId);
+
+    const QString& currentSynthResource() const;
+    void setCurrentSynthResource(const QString& newCurrentSynthResource);
 
-    void updatePluginMetaList(const VstPluginMetaList& newMetaList);
+    const QString& currentFxResource() const;
+    void setCurrentFxResource(const QString& newCurrentFxResource);
 
-    VstPluginMetaList m_pluginMetaList;
-    int m_selectedItemIndex = -1;
+signals:
+    void sequenceIdListChanged();
+    void trackIdListChanged();
+    void availableFxResourcesChanged();
+    void availableSynthResourcesChanged();
+
+    void currentSequenceIdChanged();
+    void currentTrackIdChanged();
+    void currentSynthResourceChanged();
+    void currentFxResourceChanged();
+
+private:
+    void applyNewInputParams();
+    void applyNewOutputParams();
+
+    QVariantList m_sequenceIdList;
+    QVariantList m_trackIdList;
+    QVariantList m_availableFxResources;
+    QVariantList m_availableSynthResources;
+    int m_currentSequenceId;

```

Comment:
It would be great to add default values here

Reply:
I don't care much about this dev-tools model, but fixed that one anyway